### PR TITLE
fix golint failures in pkg/kubeapiserver/options

### PR DESCRIPTION
**Pull request by yiduyangyi**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
fix golint failures in pkg/kubeapiserver/options

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubeapiserver/options/authentication.go [hunk 1]
```diff
@@ -37,13 +37,15 @@ import (
...
 	openapicommon "k8s.io/kube-openapi/pkg/common"
+
 	serviceaccountcontroller "k8s.io/kubernetes/pkg/controller/serviceaccount"
 	"k8s.io/kubernetes/pkg/features"
 	kubeauthenticator "k8s.io/kubernetes/pkg/kubeapiserver/authenticator"
 	authzmodes "k8s.io/kubernetes/pkg/kubeapiserver/authorizer/modes"
 	"k8s.io/kubernetes/plugin/pkg/auth/authenticator/token/bootstrap"
 )
 
+// BuiltInAuthenticationOptions contains all build-in authentication options for APIServer
```
Comment (sttts) [inline pkg/kubeapiserver/options/authentication.go:48]: don't we either write "API server" or "apiserver" elsewhere?

File: pkg/kubeapiserver/options/authentication.go [hunk 2]
```diff
@@ -414,16 +434,16 @@ func (s *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 }
 
 // ApplyTo requires already applied OpenAPIConfig and EgressSelector if present.
-func (o *BuiltInAuthenticationOptions) ApplyTo(authInfo *genericapiserver.AuthenticationInfo, secureServing *genericapiserver.SecureServingInfo, egressSelector *egressselector.EgressSelector, openAPIConfig *openapicommon.Config, extclient kubernetes.Interface, versionedInformer informers.SharedInformerFactory) error {
-	if o == nil {
+func (s *BuiltInAuthenticationOptions) ApplyTo(authInfo *genericapiserver.AuthenticationInfo, secureServing *genericapiserver.SecureServingInfo, egressSelector *egressselector.EgressSelector, openAPIConfig *openapicommon.Config, extclient kubernetes.Interface, versionedInformer informers.SharedInformerFactory) error {
```
Comment (sttts) [inline pkg/kubeapiserver/options/authentication.go:437]: no. These are options. So `o` is right.

File: pkg/kubeapiserver/options/cloudprovider.go [hunk 3]
```diff
@@ -20,20 +20,24 @@ import (
 	"github.com/spf13/pflag"
 )
 
+// CloudProviderOptions contains cloud provider config
 type CloudProviderOptions struct {
 	CloudConfigFile string
 	CloudProvider   string
 }
 
+// NewCloudProviderOptions create a default CloudProviderOptions
```
Comment (sttts) [inline pkg/kubeapiserver/options/cloudprovider.go:29]: creates

File: pkg/kubeapiserver/options/authentication.go (hunk 1 above)
Comment (yiduyangyi) [reply pkg/kubeapiserver/options/authentication.go:48]: I search the code, "API Server" is most used

**Pull request merged**